nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RFacetImpl.java

```diff
@@ -70,14 +71,16 @@ public Asset findOrCreateAsset(final StorageTx tx,
                                  final Map<String, String> attributes)
   {
     Bucket bucket = tx.findBucket(getRepository());
-    Asset asset = tx.findAssetWithProperty(P_NAME, path, bucket);
+    Asset asset = findAsset(tx, bucket, path);
     if (asset == null) {
       asset = tx.createAsset(bucket, component);
       asset.name(path);
-      for (Map.Entry attribute : attributes.entrySet()) {
-        asset.formatAttributes().set(attribute.getKey().toString(), attribute.getValue());
+
+      // TODO: Make this a bit more robust (could be problematic if keys are removed in later versions, or if keys clash)
+      for (Entry<String, String> attribute : attributes.entrySet()) {
+        asset.formatAttributes().set(attribute.getKey(), attribute.getValue());
       }
-      asset.formatAttributes().set(P_ASSET_KIND, getAssetKind(path));
+      asset.formatAttributes().set(P_ASSET_KIND, getAssetKind(path).name());
```

Comment:
Do we need to consider existing R assets? I think we should possibly consider an upgrade process

Reply:
You're talking about assetkind in attributes, yes? If so, master version doesn't support any kind of `.rds` metadata and this assetkind is not used on previous versions. All others are compatible with master and no need to make upgrade process.

Reply:
No, method used to return string representation before, I just decided it was a bit confusing